### DISCUSSION TITLE
[OP Stack Deployment] failed to Deploy the L1 contracts to Run a Custom Gas Token Chain

### DISCUSSION THREAD

**Discussion by draju1980** [ethereum-optimism/developers#452] [Deploying the OP Stack]
### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

I'm following _How to Run a Custom Gas Token Chain_ from https://docs.optimism.io/builders/chain-operators/management/custom-gas-token

When I run the "Deploy the L1 contracts" step's forge script command, it fails.
```
DEPLOYMENT_OUTFILE=deployments/artifact.json \
DEPLOY_CONFIG_PATH=<PATH_TO_MY_DEPLOY_CONFIG> \
  forge script scripts/Deploy.s.sol:Deploy \
  --broadcast --private-key $PRIVATE_KEY \
  --rpc-url $ETH_RPC_URL
```

kindly find full command logs on below attachment,
[Error.Deploy.the.L1.contracts.txt](https://github.com/user-attachments/files/15817848/Error.Deploy.the.L1.contracts.txt)

### Logs

== Logs ==
  Writing artifact to deployments/artifact.json
  Connected to network with chainid 11155111
  Commit hash: 363c5d7f4fb14180a0e2a28cc948fe2146f03dce
  DeployConfig: reading file deploy-config/sepolia.json
  Deploying a fresh OP Stack including SuperchainConfig
  start of L1 Deploy!
  Deploying safe: SystemOwnerSafe with salt 0x8f99807c67c946f5e140a87da1472b6871f64b31f38f4ca5b796b67d11e8683f
  Saving SafeProxyFactory: 0xa6B71E26C5e0845f74c812102Ca7114b6a896AB2
  Saving SafeSingleton: 0xd9Db270c1B5E3Bd161E8c8503c55cEABeE709552
  Saving SystemOwnerSafe: 0x0bBec9E25b224357763EB3C18F94b6D6936900f3
  New safe: SystemOwnerSafe deployed at 0x0bBec9E25b224357763EB3C18F94b6D6936900f3
    Note that this safe is owned by the deployer key
  deployed Safe!
  Setting up Superchain
  Deploying AddressManager
  Saving AddressManager: 0x5B4Dcee5079078241cF0c5FE87Cf459AAcD9c8EE
  AddressManager deployed at 0x5B4Dcee5079078241cF0c5FE87Cf459AAcD9c8EE
  Deploying ProxyAdmin
  Saving ProxyAdmin: 0x2F65B4e5a1d63Af314873935b1877507a71148aC
  ProxyAdmin deployed at 0x2F65B4e5a1d63Af314873935b1877507a71148aC
  ProxyAdmin ownership transferred to Safe at: 0x0bBec9E25b224357763EB3C18F94b6D6936900f3
  Deploying ERC1967 proxy for SuperchainConfigProxy
  Saving SuperchainConfigProxy: 0xBA0a14B802f9b2c78fa9c25BCe7fB2EF118aD59D
     at 0xBA0a14B802f9b2c78fa9c25BCe7fB2EF118aD59D
  Saving SuperchainConfig: 0x639c508D574353E47d571DE29E75246Adea1A9C8
  SuperchainConfig deployed at 0x639c508D574353E47d571DE29E75246Adea1A9C8
  Running chain assertions on the SuperchainConfig
  Deploying ERC1967 proxy for ProtocolVersionsProxy
  Saving ProtocolVersionsProxy: 0x35dD61Ab599e6B48EBe8F77961DE15190FdB78a9
     at 0x35dD61Ab599e6B48EBe8F77961DE15190FdB78a9
  Deploying ProtocolVersions implementation
  Saving ProtocolVersions: 0x8A890C9101019787111e3058559556d632A5de52
  ProtocolVersions deployed at 0x8A890C9101019787111e3058559556d632A5de52
  Running chain assertions on the ProtocolVersions
  Upgrading and initializing ProtocolVersions proxy
  ProtocolVersions version: 1.0.0
  Running chain assertions on the ProtocolVersions
  set up superchain!
  Deploying OP Chain
  Deploying proxies
  Deploying ERC1967 proxy for OptimismPortalProxy
  Saving OptimismPortalProxy: 0x1EdF731f03dc82e8f941B7DA7985eFdeCF0d797e
     at 0x1EdF731f03dc82e8f941B7DA7985eFdeCF0d797e
  Deploying ERC1967 proxy for SystemConfigProxy
  Saving SystemConfigProxy: 0xA22D51C76DA88681e2f08AFec1A5548CDF7274E6
     at 0xA22D51C76DA88681e2f08AFec1A5548CDF7274E6
  Deploying proxy for L1StandardBridge
  Saving L1StandardBridgeProxy: 0x8f44Ac46306840C0563b672Bbd774baE7e04638B
  L1StandardBridgeProxy deployed at 0x8f44Ac46306840C0563b672Bbd774baE7e04638B
  Deploying proxy for L1CrossDomainMessenger
  Saving L1CrossDomainMessengerProxy: 0xef79Af4FdecA95944F0948721de169460198A4e7
  L1CrossDomainMessengerProxy deployed at 0xef79Af4FdecA95944F0948721de169460198A4e7
  Deploying ERC1967 proxy for OptimismMintableERC20FactoryProxy
  Saving OptimismMintableERC20FactoryProxy: 0x22d3Ad67660210F49F5Cd84206aD7524367eDBa3
     at 0x22d3Ad67660210F49F5Cd84206aD7524367eDBa3
  Deploying ERC1967 proxy for L1ERC721BridgeProxy
  Saving L1ERC721BridgeProxy: 0xa663611E872800360071D475D81aE0486dCD347a
     at 0xa663611E872800360071D475D81aE0486dCD347a
  Deploying ERC1967 proxy for DisputeGameFactoryProxy
  Saving DisputeGameFactoryProxy: 0x9D123aa9b75423b6c587081306bA7D722A95D1de
     at 0x9D123aa9b75423b6c587081306bA7D722A95D1de
  Deploying ERC1967 proxy for L2OutputOracleProxy
  Saving L2OutputOracleProxy: 0xE7aF774df096cA980C4BE52D327b3a8861270EfC
     at 0xE7aF774df096cA980C4BE52D327b3a8861270EfC
  Deploying ERC1967 proxy for DelayedWETHProxy
  Saving DelayedWETHProxy: 0xF3508Df541990b813DAc10469f25aBdbB2dF1415
     at 0xF3508Df541990b813DAc10469f25aBdbB2dF1415
  Deploying ERC1967 proxy for AnchorStateRegistryProxy
  Saving AnchorStateRegistryProxy: 0xf24A24fB1371Bd7Efd2da2bDf1426635450d0Ca8
     at 0xf24A24fB1371Bd7Efd2da2bDf1426635450d0Ca8
  Transferring AddressManager ownership to ProxyAdmin
  AddressManager ownership transferred to 0x2F65B4e5a1d63Af314873935b1877507a71148aC
  Deploying implementations
  Deploying L1CrossDomainMessenger implementation
  Saving L1CrossDomainMessenger: 0x0923c7106e9Ee2f61f37792231e85601C5dbCF92
  L1CrossDomainMessenger deployed at 0x0923c7106e9Ee2f61f37792231e85601C5dbCF92
  Running chain assertions on the L1CrossDomainMessenger
  Deploying OptimismMintableERC20Factory implementation
  Saving OptimismMintableERC20Factory: 0x171875276F2E27AC9Ef7d3E23A64E09CbE99a74c
  OptimismMintableERC20Factory deployed at 0x171875276F2E27AC9Ef7d3E23A64E09CbE99a74c
  Running chain assertions on the OptimismMintableERC20Factory
  Deploying SystemConfig implementation
  Saving SystemConfig: 0x1b59609d30b7bC0e8C3a92dE9f349003F81A8E02
  SystemConfig deployed at 0x1b59609d30b7bC0e8C3a92dE9f349003F81A8E02
  Running chain assertions on the SystemConfig
  Deploying L1StandardBridge implementation
  Saving L1StandardBridge: 0x24E2e7d2608a6e75fA77958EBF4e324A0d07Ab26
  L1StandardBridge deployed at 0x24E2e7d2608a6e75fA77958EBF4e324A0d07Ab26
  Running chain assertions on the L1StandardBridge
  Deploying L1ERC721Bridge implementation
  Saving L1ERC721Bridge: 0xB9C655eAFcF2ba12D17474d315692E843458b69c
  L1ERC721Bridge deployed at 0xB9C655eAFcF2ba12D17474d315692E843458b69c
  Running chain assertions on the L1ERC721Bridge
  Deploying OptimismPortal implementation
  Saving OptimismPortal: 0x82989c4A459f884883c2d92B1A557b0269EB64c3
  OptimismPortal deployed at 0x82989c4A459f884883c2d92B1A557b0269EB64c3
  Running chain assertions on the OptimismPortal
  Guardian has no code: 0xfd1D2e729aE8eEe2E146c033bf4400fE75284301
  Deploying L2OutputOracle implementation
  Saving L2OutputOracle: 0x81e5183C2Ae760b223f52D448F6981228974EA94
  L2OutputOracle deployed at 0x81e5183C2Ae760b223f52D448F6981228974EA94
  Running chain assertions on the L2OutputOracle
  Deploying OptimismPortal2 implementation
  Saving OptimismPortal2: 0xd8d9bae8E0B12296A212FF8C962892cd9AadD943
  OptimismPortal2 deployed at 0xd8d9bae8E0B12296A212FF8C962892cd9AadD943
  Running chain assertions on the OptimismPortal2
  Guardian has no code: 0xfd1D2e729aE8eEe2E146c033bf4400fE75284301
  Deploying DisputeGameFactory implementation
  Saving DisputeGameFactory: 0xBCd27d394C5fD6C9403309c2C29229c685ce8f4D
  DisputeGameFactory deployed at 0xBCd27d394C5fD6C9403309c2C29229c685ce8f4D
  Running chain assertions on the DisputeGameFactory
  Deploying DelayedWETH implementation
  Saving DelayedWETH: 0xA2A913d092Faba9a63A9Ba83E22CA7ca10AC2be4
  DelayedWETH deployed at 0xA2A913d092Faba9a63A9Ba83E22CA7ca10AC2be4
  Running chain assertions on the DelayedWETH
  Deploying PreimageOracle implementation
  Saving PreimageOracle: 0x47D52B36aD1cD764818eBdabb934Db9cD7F0b4EC
  PreimageOracle deployed at 0x47D52B36aD1cD764818eBdabb934Db9cD7F0b4EC
  Deploying Mips implementation
  Saving Mips: 0x7392A0Ee4F826A99CE273c6044C82DAF541F9278
  MIPS deployed at 0x7392A0Ee4F826A99CE273c6044C82DAF541F9278
  Deploying AnchorStateRegistry implementation
  Saving AnchorStateRegistry: 0x6549C94f9e763a5d32906C75F816DCE8cfeD9161
  AnchorStateRegistry deployed at 0x6549C94f9e763a5d32906C75F816DCE8cfeD9161
  Initializing implementations
  Fault proofs enabled. Initializing the OptimismPortal proxy with the OptimismPortal2.
  Upgrading and initializing OptimismPortal2 proxy
  OptimismPortal2 version: 3.10.0
  Running chain assertions on the OptimismPortal2
  Guardian has no code: 0xfd1D2e729aE8eEe2E146c033bf4400fE75284301
  Upgrading and initializing SystemConfig proxy
```
Error:
script failed: revert: GS013
```

**Comment by opfocus** [comment]
CC @Chomtana  CC @sbvegan

**Comment by priyanshuthapliyal55** [comment]
> ### Did you check the documentation?
> * [x]  Yes
> 
> ### Did you check for duplicate questions?
> * [x]  Yes
> 
> ### Issue Description
> I'm following _How to Run a Custom Gas Token Chain_ from https://docs.optimism.io/builders/chain-operators/management/custom-gas-token
> 
> When I run the "Deploy the L1 contracts" step's forge script command, it fails.
> 
> ```
> DEPLOYMENT_OUTFILE=deployments/artifact.json \
> DEPLOY_CONFIG_PATH=<PATH_TO_MY_DEPLOY_CONFIG> \
>   forge script scripts/Deploy.s.sol:Deploy \
>   --broadcast --private-key $PRIVATE_KEY \
>   --rpc-url $ETH_RPC_URL
> ```
> 
> kindly find full command logs on below attachment, [Error.Deploy.the.L1.contracts.txt](https://github.com/user-attachments/files/15817848/Error.Deploy.the.L1.contracts.txt)
> 
> ### Logs
> == Logs == Writing artifact to deployments/artifact.json Connected to network with chainid 11155111 Commit hash: 363c5d7f4fb14180a0e2a28cc948fe2146f03dce DeployConfig: reading file deploy-config/sepolia.json Deploying a fresh OP Stack including SuperchainConfig start of L1 Deploy! Deploying safe: SystemOwnerSafe with salt 0x8f99807c67c946f5e140a87da1472b6871f64b31f38f4ca5b796b67d11e8683f Saving SafeProxyFactory: 0xa6B71E26C5e0845f74c812102Ca7114b6a896AB2 Saving SafeSingleton: 0xd9Db270c1B5E3Bd161E8c8503c55cEABeE709552 Saving SystemOwnerSafe: 0x0bBec9E25b224357763EB3C18F94b6D6936900f3 New safe: SystemOwnerSafe deployed at 0x0bBec9E25b224357763EB3C18F94b6D6936900f3 Note that this safe is owned by the deployer key deployed Safe! Setting up Superchain Deploying AddressManager Saving AddressManager: 0x5B4Dcee5079078241cF0c5FE87Cf459AAcD9c8EE AddressManager deployed at 0x5B4Dcee5079078241cF0c5FE87Cf459AAcD9c8EE Deploying ProxyAdmin Saving ProxyAdmin: 0x2F65B4e5a1d63Af314873935b1877507a71148aC ProxyAdmin deployed at 0x2F65B4e5a1d63Af314873935b1877507a71148aC ProxyAdmin ownership transferred to Safe at: 0x0bBec9E25b224357763EB3C18F94b6D6936900f3 Deploying ERC1967 proxy for SuperchainConfigProxy Saving SuperchainConfigProxy: 0xBA0a14B802f9b2c78fa9c25BCe7fB2EF118aD59D at 0xBA0a14B802f9b2c78fa9c25BCe7fB2EF118aD59D Saving SuperchainConfig: 0x639c508D574353E47d571DE29E75246Adea1A9C8 SuperchainConfig deployed at 0x639c508D574353E47d571DE29E75246Adea1A9C8 Running chain assertions on the SuperchainConfig Deploying ERC1967 proxy for ProtocolVersionsProxy Saving ProtocolVersionsProxy: 0x35dD61Ab599e6B48EBe8F77961DE15190FdB78a9 at 0x35dD61Ab599e6B48EBe8F77961DE15190FdB78a9 Deploying ProtocolVersions implementation Saving ProtocolVersions: 0x8A890C9101019787111e3058559556d632A5de52 ProtocolVersions deployed at 0x8A890C9101019787111e3058559556d632A5de52 Running chain assertions on the ProtocolVersions Upgrading and initializing ProtocolVersions proxy ProtocolVersions version: 1.0.0 Running chain assertions on the ProtocolVersions set up superchain! Deploying OP Chain Deploying proxies Deploying ERC1967 proxy for OptimismPortalProxy Saving OptimismPortalProxy: 0x1EdF731f03dc82e8f941B7DA7985eFdeCF0d797e at 0x1EdF731f03dc82e8f941B7DA7985eFdeCF0d797e Deploying ERC1967 proxy for SystemConfigProxy Saving SystemConfigProxy: 0xA22D51C76DA88681e2f08AFec1A5548CDF7274E6 at 0xA22D51C76DA88681e2f08AFec1A5548CDF7274E6 Deploying proxy for L1StandardBridge Saving L1StandardBridgeProxy: 0x8f44Ac46306840C0563b672Bbd774baE7e04638B L1StandardBridgeProxy deployed at 0x8f44Ac46306840C0563b672Bbd774baE7e04638B Deploying proxy for L1CrossDomainMessenger Saving L1CrossDomainMessengerProxy: 0xef79Af4FdecA95944F0948721de169460198A4e7 L1CrossDomainMessengerProxy deployed at 0xef79Af4FdecA95944F0948721de169460198A4e7 Deploying ERC1967 proxy for OptimismMintableERC20FactoryProxy Saving OptimismMintableERC20FactoryProxy: 0x22d3Ad67660210F49F5Cd84206aD7524367eDBa3 at 0x22d3Ad67660210F49F5Cd84206aD7524367eDBa3 Deploying ERC1967 proxy for L1ERC721BridgeProxy Saving L1ERC721BridgeProxy: 0xa663611E872800360071D475D81aE0486dCD347a at 0xa663611E872800360071D475D81aE0486dCD347a Deploying ERC1967 proxy for DisputeGameFactoryProxy Saving DisputeGameFactoryProxy: 0x9D123aa9b75423b6c587081306bA7D722A95D1de at 0x9D123aa9b75423b6c587081306bA7D722A95D1de Deploying ERC1967 proxy for L2OutputOracleProxy Saving L2OutputOracleProxy: 0xE7aF774df096cA980C4BE52D327b3a8861270EfC at 0xE7aF774df096cA980C4BE52D327b3a8861270EfC Deploying ERC1967 proxy for DelayedWETHProxy Saving DelayedWETHProxy: 0xF3508Df541990b813DAc10469f25aBdbB2dF1415 at 0xF3508Df541990b813DAc10469f25aBdbB2dF1415 Deploying ERC1967 proxy for AnchorStateRegistryProxy Saving AnchorStateRegistryProxy: 0xf24A24fB1371Bd7Efd2da2bDf1426635450d0Ca8 at 0xf24A24fB1371Bd7Efd2da2bDf1426635450d0Ca8 Transferring AddressManager ownership to ProxyAdmin AddressManager ownership transferred to 0x2F65B4e5a1d63Af314873935b1877507a71148aC Deploying implementations Deploying L1CrossDomainMessenger implementation Saving L1CrossDomainMessenger: 0x0923c7106e9Ee2f61f37792231e85601C5dbCF92 L1CrossDomainMessenger deployed at 0x0923c7106e9Ee2f61f37792231e85601C5dbCF92 Running chain assertions on the L1CrossDomainMessenger Deploying OptimismMintableERC20Factory implementation Saving OptimismMintableERC20Factory: 0x171875276F2E27AC9Ef7d3E23A64E09CbE99a74c OptimismMintableERC20Factory deployed at 0x171875276F2E27AC9Ef7d3E23A64E09CbE99a74c Running chain assertions on the OptimismMintableERC20Factory Deploying SystemConfig implementation Saving SystemConfig: 0x1b59609d30b7bC0e8C3a92dE9f349003F81A8E02 SystemConfig deployed at 0x1b59609d30b7bC0e8C3a92dE9f349003F81A8E02 Running chain assertions on the SystemConfig Deploying L1StandardBridge implementation Saving L1StandardBridge: 0x24E2e7d2608a6e75fA77958EBF4e324A0d07Ab26 L1StandardBridge deployed at 0x24E2e7d2608a6e75fA77958EBF4e324A0d07Ab26 Running chain assertions on the L1StandardBridge Deploying L1ERC721Bridge implementation Saving L1ERC721Bridge: 0xB9C655eAFcF2ba12D17474d315692E843458b69c L1ERC721Bridge deployed at 0xB9C655eAFcF2ba12D17474d315692E843458b69c Running chain assertions on the L1ERC721Bridge Deploying OptimismPortal implementation Saving OptimismPortal: 0x82989c4A459f884883c2d92B1A557b0269EB64c3 OptimismPortal deployed at 0x82989c4A459f884883c2d92B1A557b0269EB64c3 Running chain assertions on the OptimismPortal Guardian has no code: 0xfd1D2e729aE8eEe2E146c033bf4400fE75284301 Deploying L2OutputOracle implementation Saving L2OutputOracle: 0x81e5183C2Ae760b223f52D448F6981228974EA94 L2OutputOracle deployed at 0x81e5183C2Ae760b223f52D448F6981228974EA94 Running chain assertions on the L2OutputOracle Deploying OptimismPortal2 implementation Saving OptimismPortal2: 0xd8d9bae8E0B12296A212FF8C962892cd9AadD943 OptimismPortal2 deployed at 0xd8d9bae8E0B12296A212FF8C962892cd9AadD943 Running chain assertions on the OptimismPortal2 Guardian has no code: 0xfd1D2e729aE8eEe2E146c033bf4400fE75284301 Deploying DisputeGameFactory implementation Saving DisputeGameFactory: 0xBCd27d394C5fD6C9403309c2C29229c685ce8f4D DisputeGameFactory deployed at 0xBCd27d394C5fD6C9403309c2C29229c685ce8f4D Running chain assertions on the DisputeGameFactory Deploying DelayedWETH implementation Saving DelayedWETH: 0xA2A913d092Faba9a63A9Ba83E22CA7ca10AC2be4 DelayedWETH deployed at 0xA2A913d092Faba9a63A9Ba83E22CA7ca10AC2be4 Running chain assertions on the DelayedWETH Deploying PreimageOracle implementation Saving PreimageOracle: 0x47D52B36aD1cD764818eBdabb934Db9cD7F0b4EC PreimageOracle deployed at 0x47D52B36aD1cD764818eBdabb934Db9cD7F0b4EC Deploying Mips implementation Saving Mips: 0x7392A0Ee4F826A99CE273c6044C82DAF541F9278 MIPS deployed at 0x7392A0Ee4F826A99CE273c6044C82DAF541F9278 Deploying AnchorStateRegistry implementation Saving AnchorStateRegistry: 0x6549C94f9e763a5d32906C75F816DCE8cfeD9161 AnchorStateRegistry deployed at 0x6549C94f9e763a5d32906C75F816DCE8cfeD9161 Initializing implementations Fault proofs enabled. Initializing the OptimismPortal proxy with the OptimismPortal2. Upgrading and initializing OptimismPortal2 proxy OptimismPortal2 version: 3.10.0 Running chain assertions on the OptimismPortal2 Guardian has no code: 0xfd1D2e729aE8eEe2E146c033bf4400fE75284301 Upgrading and initializing SystemConfig proxy
> 
> ```
> Error:
> script failed: revert: GS013
> ```
> 
> ### Additional Information
> _No response_
> 
> ### Feedback
> _No response_

did you set direnv allow command after using this command cp .envrc.example .envrc ? also  if you have set fault proof = true it wont work with custom gas token

**Reply by draju1980** [reply]
Yes, I have loaded the variables with direnv, but I have not set FAULT_PROOF=true in `.envrc` or `deploy-config/sepolia.json`. 

```
➜  contracts-bedrock git:(op-contracts/v2.0.0-beta.2) ✗ direnv allow
direnv: loading ~/optimism/.envrc
direnv: export +DEPLOYMENT_CONTEXT +ETHERSCAN_API_KEY +ETH_RPC_URL +GS_ADMIN_ADDRESS +GS_ADMIN_PRIVATE_KEY +GS_BATCHER_ADDRESS +GS_BATCHER_PRIVATE_KEY +GS_PROPOSER_ADDRESS +GS_PROPOSER_PRIVATE_KEY +GS_SEQUENCER_ADDRESS +GS_SEQUENCER_PRIVATE_KEY +IMPL_SALT +L1_RPC_KIND +L1_RPC_URL +PRIVATE_KEY +TENDERLY_PROJECT +TENDERLY_USERNAME
➜  contracts-bedrock git:(op-contracts/v2.0.0-beta.2) ✗
```

**Reply by priyanshuthapliyal** [reply]
> Yes, I have loaded the variables with direnv, but I have not set FAULT_PROOF=true in `.envrc` or `deploy-config/sepolia.json`.
> 
> ```
> ➜  contracts-bedrock git:(op-contracts/v2.0.0-beta.2) ✗ direnv allow
> direnv: loading ~/optimism/.envrc
> direnv: export +DEPLOYMENT_CONTEXT +ETHERSCAN_API_KEY +ETH_RPC_URL +GS_ADMIN_ADDRESS +GS_ADMIN_PRIVATE_KEY +GS_BATCHER_ADDRESS +GS_BATCHER_PRIVATE_KEY +GS_PROPOSER_ADDRESS +GS_PROPOSER_PRIVATE_KEY +GS_SEQUENCER_ADDRESS +GS_SEQUENCER_PRIVATE_KEY +IMPL_SALT +L1_RPC_KIND +L1_RPC_URL +PRIVATE_KEY +TENDERLY_PROJECT +TENDERLY_USERNAME
> ➜  contracts-bedrock git:(op-contracts/v2.0.0-beta.2) ✗
> ```

in your logs i can see that fault proof enable may be you can try mentioning "useFaultProofs": false in deploy-config itself.

**Reply by draju1980** [reply]
Thanks, @priyanshuthapliyal. I have updated the useFaultProofs argument in deploy-config and re-run the command, but I am still getting an error. For your reference, I am sending the command output logs here.
[Error deploy L1 contracts.txt](https://github.com/user-attachments/files/15832320/Error.deploy.L1.contracts.txt)

**Reply by kyritzb** [reply]
I am running into the exact same issues you are having. I am actively looking for a fix to this or seeing if anyone else has a fix. If you figure it out pls respond :)

**Comment by thekera404** [comment]
good

**Comment by tynes** [comment]
```
Error:
script failed: revert: GS013
```
This error comes from gnosis safe, need to investigate based on that codebase

**Reply by draju1980** [reply]
FYI, I used the `v2.0.0-beta.2` branch.

**Comment by sbvegan** [comment]
Hey @draju1980, as @tynes noted, this does look like a gnosis safe error. It would also help to know what parameters you used for the deploy configuration. Can you post the contents in the thread?

**Reply by draju1980** [reply]
Hello @sbvegan @tynes ,  here are the deploy config parameters,
```
{
  "finalSystemOwner": "0xfd1D2e729aE8eEe2E146c033bf4400fE75284301",
  "superchainConfigGuardian": "0xfd1D2e729aE8eEe2E146c033bf4400fE75284301",
  "l1StartingBlockTag": "0x48f520cf4ddaf34c8336e6e490632ea3cf1e5e93b0b2bc6e917557e31845371b",
  "l1ChainID": 11155111,
  "l2ChainID": 11155420,
  "l2BlockTime": 2,
  "l1BlockTime": 12,
  "maxSequencerDrift": 600,
  "sequencerWindowSize": 3600,
  "channelTimeout": 300,
  "p2pSequencerAddress": "0x57CACBB0d30b01eb2462e5dC940c161aff3230D3",
  "batchInboxAddress": "0xff00000000000000000000000000000011155420",
  "batchSenderAddress": "0x8F23BB38F531600e5d8FDDaAEC41F13FaB46E98c",
  "l2OutputOracleSubmissionInterval": 120,
  "l2OutputOracleStartingBlockNumber": 0,
  "l2OutputOracleStartingTimestamp": 1690493568,
  "l2OutputOracleProposer": "0x49277EE36A024120Ee218127354c4a3591dc90A9",
  "l2OutputOracleChallenger": "0xfd1D2e729aE8eEe2E146c033bf4400fE75284301",
  "finalizationPeriodSeconds": 12,
  "proxyAdminOwner": "0xfd1D2e729aE8eEe2E146c033bf4400fE75284301",
  "baseFeeVaultRecipient": "0xfd1D2e729aE8eEe2E146c033bf4400fE75284301",
  "l1FeeVaultRecipient": "0xfd1D2e729aE8eEe2E146c033bf4400fE75284301",
  "sequencerFeeVaultRecipient": "0xfd1D2e729aE8eEe2E146c033bf4400fE75284301",
  "baseFeeVaultMinimumWithdrawalAmount": "0x8ac7230489e80000",
  "l1FeeVaultMinimumWithdrawalAmount": "0x8ac7230489e80000",
  "sequencerFeeVaultMinimumWithdrawalAmount": "0x8ac7230489e80000",
  "baseFeeVaultWithdrawalNetwork": 0,
  "l1FeeVaultWithdrawalNetwork": 0,
  "sequencerFeeVaultWithdrawalNetwork": 0,
  "enableGovernance": true,
  "governanceTokenSymbol": "OP",
  "governanceTokenName": "Optimism",
  "governanceTokenOwner": "0xfd1D2e729aE8eEe2E146c033bf4400fE75284301",
  "l2GenesisBlockGasLimit": "0x1c9c380",
  "l2GenesisBlockBaseFeePerGas": "0x3b9aca00",
  "eip1559Denominator": 50,
  "eip1559Elasticity": 6,
  "l2GenesisRegolithTimeOffset": "0x0",
  "systemConfigStartBlock": 4071248,
  "requiredProtocolVersion": "0x0000000000000000000000000000000000000004000000000000000000000001",
  "recommendedProtocolVersion": "0x0000000000000000000000000000000000000004000000000000000000000001",
  "fundDevAccounts": false,
  "faultGameAbsolutePrestate": "0x03e69d3de5155f4a80da99dd534561cbddd4f9dd56c9ecc704d6886625711d2b",
  "faultGameMaxDepth": 73,
  "faultGameClockExtension": 10800,
  "faultGameMaxClockDuration": 302400,
  "faultGameGenesisBlock": 9496192,
  "faultGameGenesisOutputRoot": "0x63b1cda487c072b020a57c1203f7c2921754005cadbd54bed7f558111b8278d8",
  "faultGameSplitDepth": 30,
  "faultGameWithdrawalDelay": 604800,
  "preimageOracleMinProposalSize": 126000,
  "preimageOracleChallengePeriod": 86400,
  "proofMaturityDelaySeconds": 604800,
  "disputeGameFinalityDelaySeconds": 302400,
  "respectedGameType": 0,
  "useFaultProofs": false,
  "baseFeeVaultWithdrawalNetwork": 1,
  "l1FeeVaultWithdrawalNetwork": 1,
  "sequencerFeeVaultWithdrawalNetwork": 1,
  "baseFeeVaultRecipient": "0x48b142E11a2D2CfCA1C59d5F15b5A040E1b7aDaE",
  "l1FeeVaultRecipient": "0x48b142E11a2D2CfCA1C59d5F15b5A040E1b7aDaE",
  "sequencerFeeVaultRecipient": "0x48b142E11a2D2CfCA1C59d5F15b5A040E1b7aDaE",
  "useCustomGasToken": true,
  "customGasTokenAddress": "0xcC671621d143E04Dd88440EFCe798e3d87E6Ab5A"
}
```

**Reply by kyritzb** [reply]
did you ever end up figuring it out?

**Reply by draju1980** [reply]
I haven't resolved the issue yet, despite trying all the options based on the OP documentation. I've since been occupied with other projects. However, resolving this error would be extremely helpful.

**Reply by kyritzb** [reply]
I'll message here if I end up getting it, I have also been stuck and focusing on other aspects of development hoping this is resolved. Getting the L2 up with no custom gas token wasn't too bad, but with a custom gas token has been a struggle.

I got the same exact errors as you.

**Reply by opfocus** [reply]
https://github.com/ethereum-optimism/specs/discussions/140   Please also follow this thread

**Reply by kyritzb** [reply]
Thank you for sharing.

**Reply by opfocus** [reply]
Is this Sepolia’s config?  you need your own config https://docs.optimism.io/builders/chain-operators/tutorials/create-l2-rollup#configure-your-network
Generated config.json may not apply directly to you, you may need to manually add some missing configuration after generated

**Comment by kyritzb** [comment]
Hi all! I'm still looking for a fix. Will take another stab at it and see if I can get it to work. If anyone has solved it please let me know.
